File: dsa/2-bigO/linear-time-O(n)/linearTime.js
const groceries = ["milk", "bread", "eggs", "flour", "cheese", "sugar"];

const searchForItem = (item) => {
    for (let i = 0; i < groceries.length; i++) {
        if (groceries[i] === item) {
        console.log(`Found ${item}`);
        }
    }
};
searchForItem("eggs");
// O(n)


const searchFor2Items = (item) => {
    for (let i = 0; i < groceries.length; i++) {
        if (groceries[i] === item) {
        console.log(`Found ${item}`);
        }
    }

    for (let j = 0; j < groceries.length; j++) {
        if (groceries[j] === item) {
        console.log(`Found ${item} 2`);
        }
    }

};
searchFor2Items("eggs");
// O(n + n) = O(2n) = O(n)
// Drop the constant so it becomes O(n) significa ignorar constantes multiplicativas al analizar la eficiencia de un algoritmo, porque lo importante es cómo escala con el tamaño del input (n), no cuántas veces exactamente se ejecuta.

File: dsa/2-bigO/a-linearTime/linear-time.py
groceries = ["milk", "bread", "eggs", "flour", "cheese", "sugar"]


def search_for_item(item):
    for i in groceries:
        if groceries[i] == item:
            print(f"Found {item}")


search_for_item("eggs")
# O(n)


def search_for_2_items(item):
    for i in groceries:
        if groceries[i] == item:
            print(f"Found {item}")

    for j in groceries:
        if groceries[j] == item:
            print(f"Found {item}")


search_for_2_items("eggs")
# O(n + n) = O(2n) = O(n)
# Drop the constant so it becomes O(n) significa ignorar constantes multiplicativas al analizar la eficiencia de un algoritmo, porque lo importante es cómo escala con el tamaño del input (n), no cuántas veces exactamente se ejecuta.
